doc/rtd/topics/datasources/nocloud.rst

```diff
@@ -32,15 +32,84 @@ The permitted keys are:
 
 With ``ds=nocloud``, the ``seedfrom`` value must start with ``/`` or
 ``file://``.  With ``ds=nocloud-net``, the ``seedfrom`` value must start
-with ``http://`` or ``https://``.
+with ``http://`` or ``https://`` and end with a trailing ``/``.
+
+Cloud-init performs variable expansion of the seedfrom URL for any DMI kernel
+variables present in ``/sys/class/dmi/id`` or FreeBSD's kenv.
+Your ``seedfrom`` URL can contain variable names of the format `:dmi.varname:`
+to indicate to cloud-init NoCloud datasource that variable expansion to the
+actual value of the DMI system attribute is wanted. The leading and trailing
+colon `:` before and after any `:dmi.<varname>:` are used because other
+metacharacters would hae to be escaped to avoid Grub interpreting those chars.
+Typically you want to use cloud-init's distro-agnostic alias
+
+.. list-table:: Available DMI variables for expansion in ``seedfrom`` URL
+  :widths: 25 25 50
+  :header-rows: 1
+
+  * - Distro-agnostic DMI string
```

Comment:
Nice formatting.

What does `Distro-agnostic` mean in this context? Linux distro? If it's linux distro, then what's the difference between this column and the `Linux-specific` column? If it's encompassing of FreeBSD, then why do we need multiple columns instead of just the `Distro-agnostic` list?

Reply:
@TheRealFalcon we probably don't. I was documenting them for reference. All aliases are allowed for those familiar with either the naming of the sysfs dmi file IDs or the kenv vars. I didn't want to preclude use of those keys because they will inevitably may be more familiar than the DMI standard strings presented by `dmidecode -s`.  Maybe this is too much to document though or maybe we should actually limit the aliases supported to only the standardized DMI string names.
What do you think?

Reply:
As discussed in IRC. We don't really need/want distro-specific options as DMI standard strings will provide an alias that works on both FreeBSD and Linux. Folks can look to `dmidecode -s` for references of keys on their systems regardless of *BSD or Linux.